### PR TITLE
Drupal API - Custom User create endpoint

### PR DESCRIPTION
**Pull request by aaronschachter**
@angaither Can you pls review? cc: @DeeZone 

Adds custom resource at `api/v1/users` via `dosomething_user_services_resources` to allow for easy account importing.  

Right now this endpoint is available to anonymous users to unblock @DeeZone as we still sort out the best way to handle API validation.  I will update the API wiki once PR is merged.

Sample request: **POST** http://dev.dosomething.org:8888/api/v1/users 

with posted data:

```
{
"email":"aschachter+fugazi+123@dosomething.org",
"password":"password",
"birthdate":"1995-12-31",
"first_name": "Aa Dog",
"user_registration_source":"services"
}
```

Example response:

```
{
    "mail": "aschachter+fugazi+123@dosomething.org",
    "name": "1700339",
    "status": 1,
    "created": 1406582002,
    "pass": "$S$D.zLO5IJbVnWo7NFgMDAJSiIYs8Pl/mak.3S7agLOR4f/8u0.Y5s",
    "field_birthdate": {
        "und": [{
            "value": "1995-12-31",
            "timezone": "UTC",
            "timezone_db": "UTC",
            "date_type": "datetime"
        }]
    },
    "field_first_name": {
        "und": [{
            "value": "Aa Dog"
        }]
    },
    "field_user_registration_source": {
        "und": [{
            "value": "services"
        }]
    },
    "uuid": "530903a1-f87d-4f22-bf84-d9f8d6edf227",
    "uid": "1700339",
    "theme": "",
    "signature": "",
    "access": 0,
    "login": 0,
    "language": "",
    "picture": 0,
    "init": "",
    "roles": {
        "2": "authenticated user"
    },
    "field_under_thirteen": {
        "und": [{
            "value": 0
        }]
    },
    "field_address": {
        "und": [{
            "element_key": "user|user|field_address|und|0",
            "thoroughfare": "",
            "premise": "",
            "locality": "",
            "administrative_area": "",
            "postal_code": "",
            "country": "US"
        }]
    }
}
```

### CODE REVIEW

**Comment by deezone** [comment]
Fantastic! :+1:

File: lib/modules/dosomething/dosomething_user/resources/dosomething_user.resource.inc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+function _dosomething_user_resource_create($account) {
+  if (!isset($account['email'])) {
+    return services_error("Email is required.");
+  }
+  // Initialize array to pass to user_save().
+  $edit = array();
+  $edit['mail'] = $account['email'];
+  $edit['name'] = $account['email'];
+  $edit['status'] = 1;
+  $edit['created'] = REQUEST_TIME;
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_user/resources/dosomething_user.resource.inc:44]: Consider setting roles, [example](http://codekarate.com/blog/create-user-account-drupal-7-programmatically).

File: lib/modules/dosomething/dosomething_user/dosomething_user.module [hunk 2]
```diff
@@ -79,6 +79,36 @@ function dosomething_user_permission() {
...
+            array(
+              'name' => 'account',
+              'type' => 'array',
+              'description' => 'The user object',
+              'source' => 'data',
+              'optional' => FALSE,
+            ),
+          ),
+        'access callback' => '_dosomething_user_resource_access',
+      ),
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_user/dosomething_user.module:106]: I'm not super familiar with create a user service, but do we need to put some sort of permission on this? Or token validation?

File: lib/modules/dosomething/dosomething_user/resources/dosomething_user.resource.inc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * @see dosomething_user_services_resources()
+ */
+
+/**
+ * Access callback for User resources.
+ */
+function _dosomething_user_resource_access() {
+  // Return TRUE for development.
+  return TRUE;
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_user/resources/dosomething_user.resource.inc:14]: Ah, maybe this is that?

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_user/resources/dosomething_user.resource.inc:14]: Yup, this is the access callback.  For now this resource is able to be called by anonymous requests, as the User Import this is for is high priority and our repo isn't public yet.

**Pull request merged**